provider/management/commands/delete_expired_grant_tokens.py

```diff
@@ -50,34 +50,29 @@ def handle(self, *args, **options):
         if sleep_between < 0:
             raise CommandError('Only non-negative sleep between seconds is allowed ({}).'.format(sleep_between))
 
-        table_name = 'oauth2_grant'
         delete_date = datetime.now()
-        where_clause = "WHERE expires < '{}'".format(delete_date.strftime('%Y-%m-%d %H:%m:%S'))
-        # The "as id" below fools Django raw query into thinking the primary key is being queried.
-        total_expired_tokens = Grant.objects.raw(
-            'SELECT COUNT(*) as id FROM {} {}'.format(table_name, where_clause)
-        )[0].id
-        log.info("STARTED: Deleting %s expired grant tokens older than '%s' with chunk size of %s and %s seconds between chunk.",
+        total_expired_tokens = Grant.objects.filter(expires__lte=delete_date).count()
+
+        log.info(
+            "STARTED: Deleting %s expired grant tokens older than '%s' w/ chunk size of %s and %s secs between chunks",
             total_expired_tokens, delete_date, chunk_size, sleep_between
-        )
+            )
 
         total_deletions = 0
-        while total_deletions < total_expired_tokens:
-            deletions_now = min(chunk_size, total_expired_tokens - total_deletions)
+
+        while True:
+            target_ids = Grant.objects.filter(expires__lte=delete_date).values_list('id', flat=True)[:chunk_size]
```

Comment:
What are we optimizing for with chunks? This just seems like we are locking the able multiple times rather than once.

Reply:
deleting hundreds of thousands of rows at once causes EBS to freak out.  We used to chunk with LIMIT below (the subclause of the DELETE).  In production we like to run commands like this as "delete 2000 rows and sleep 5 seconds" to stop the DB from freaking out when the disk all goes to the delete.

Reply:
What @jibsheet matches my experience. This goes into some additional detail as to why: http://mysql.rjweb.org/doc.php/deletebig